From ac21f2da9b60289acbf58c6d03acf988c68aeca1 Mon Sep 17 00:00:00 2001
From: Marek Majtyka <marekx.majtyka@intel.com>
Date: Fri, 3 Nov 2017 10:48:30 -0500
Subject: [PATCH 257/278] drivers/power/reset: Fix Axxia Retention Reset Issue

Retention reset trigger shall only update syscon scratch register
particular flag instead of overwriting it. Some fatal edac errors
might be lost.

Signed-off-by: Marek Majtyka <marekx.majtyka@intel.com>
---
 drivers/power/reset/axxia-reset.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/reset/axxia-reset.c b/drivers/power/reset/axxia-reset.c
index 195d7cd..60afa8e 100644
--- a/drivers/power/reset/axxia-reset.c
+++ b/drivers/power/reset/axxia-reset.c
@@ -54,7 +54,7 @@ static void (*saved_arm_pm_restart)
 		return;
 
 	/* set retention reset bit in pscratch */
-	regmap_write(syscon, SC_PSCRATCH, 1);
+	regmap_update_bits(syscon, SC_PSCRATCH, 1, 1);
 
 	/* trap into secure monitor to do the reset */
 #ifdef CONFIG_POWER_RESET_AXXIA_DDR_RETENTION
-- 
1.8.5.2.233.g932f7e4

